Docs: Update CloudWatch and Stackdriver docs for 6.6 (#21679)
grafana/grafana: docs/sources/features/datasources/stackdriver.md

Group by resource or metric labels to reduce the number of time series and to aggregate the results by a group by. E.g. Group by instance_name to see an aggregated metric for a Compute instance.

#### Meta data labels

> Meta data labels are only returned in the time series response if they're part of the group by segment in the time series request. There's no API for retrieving meta data labels, and therefore it's not possible to populate the group by dropdown with the meta data labels that are avaiable for the selected service and metric. However, the group by field dropdown comes with a pre-defined list of common system labels. User labels cannot be pre-defined, but it's possible to enter them manually in the group by field. If a meta data label, user label or system label, is included in the group by segment, it will be possible to create filters based on it and to expand its value on the alias field.

Reviewer: I think the term `group by segment` might be hard to parse for most people. I wouldn't use the word segment. Generally this paragraph is very technical and almost assumes that people know about the underlying API. Maybe @oddlittlebird can help?

Reviewer 2: "meta data" is one word: metadata

Group by resource or metric labels to reduce the number of time series and to aggregate the results by a group by. E.g. Group by instance_name to see an aggregated metric for a Compute instance.

> #### Metadata labels

Reviewer 2: This section is much easier to read now!

Author: Nice! Do you want to approve so that I can merge?

#### Meta data labels

> Meta data labels are only returned in the time series response if they're part of the group by segment in the time series request. There's no API for retrieving meta data labels, and therefore it's not possible to populate the group by dropdown with the meta data labels that are avaiable for the selected service and metric. However, the group by field dropdown comes with a pre-defined list of common system labels. User labels cannot be pre-defined, but it's possible to enter them manually in the group by field. If a meta data label, user label or system label, is included in the group by segment, it will be possible to create filters based on it and to expand its value on the alias field.

Reviewer 2: ```suggestion
Metadata labels are only returned in the time series response if they're part of the group by segment in the time series request. There's no API for retrieving metadata labels, so it's not possible to populate the group by dropdown with the metadata labels that are available for the selected service and metric. However, the group by field dropdown comes with a pre-defined list of common system labels. 

User labels cannot be pre-defined, but it's possible to enter them manually in the group by field. If a metadata label, user label or system label is included in the group by segment, then you can create filters based on it and expand its value on the alias field.
```

Reviewer 2: A few more things:

It looks like you might be talking about field names in some cases. If so, capitalize them to match what is on the screen and make them bold, per the documentation style guide.

Most of this paragraph seems to assume the reader understands the API and the metadata labels, which I unfortunately do not, so it is a bit dense for me to clearly understand what you are trying to say here.
